common/remote/rmtsmtp.cpp

```diff
@@ -549,15 +560,15 @@ class CMailInfo
     {
         try
         {
-            socket->read(inbuff,1,sizeof(inbuff),inlen);
+            socket->readtms(inbuff, 1, sizeof(inbuff), inlen, 30000);
+
             //MORE: the following is somewhat primitive and not RFC compliant (see bug 25951) - but it is a lot better than nothing
             if((*inbuff == '4') || (*inbuff == '5'))
             {
                 StringBuffer b;
-                b.append("Negative reply from mail server at ").append(mailServer.get()).append(":").append(port).append(" after writing ").append(lastAction.str()).append(" in SendEmail*: ").append(inlen, inbuff).clip();
-                WARNLOG("%s", b.str());
-                if (warnings)
-                    warnings->append(b.str());
+                b.append("Exception: negative reply from mail server at ").append(mailServer.get()).append(":").append(port).append(" after writing ").append(lastAction.str()).append(" in SendEmail*: ").append(inlen, inbuff).clip();
+                // don't continue on after these responses ...
+                throw makeStringException(MSGAUD_operator, 0, b.str());
```

Comment:
A few questions.
a) Will this now fail a job if the email server doesn't respond?  I think so, in which case this is a bit dangerous.  It should probably be caught and added to warnings in doSendEmail.
b) Should we throw different errors for 4xx and 5xx, with a retry loop in doSendEmail for 4xx codes?
c) This throw exception will be caught and rethrown in a different format.  It may be better to check outside the try block and throw there.

Reply:
Reworked PR to improve error handling.  Please re-review when you get a chance.